droidmediacodec.cpp

```diff
@@ -766,16 +766,38 @@ bool droid_media_codec_is_supported(DroidMediaCodecMetaData *meta, bool encoder)
     return matchingCodecs.size() > 0;
 }
 
-unsigned int droid_media_codec_get_supported_color_formats(const char *mime, int encoder, uint32_t *formats, unsigned int maxFormats)
+unsigned int droid_media_codec_get_supported_color_formats(DroidMediaCodecMetaData *meta, int encoder, uint32_t *formats, unsigned int maxFormats)
 {
+#if ANDROID_MAJOR >= 5
     android::sp<android::MediaCodecList::IMediaCodecList> list = android::MediaCodecList::getInstance();
-    int index = 0;
-    ssize_t matchIndex = list->findCodecByType(mime, encoder, index);
+#else
+    const android::MediaCodecList *list = android::MediaCodecList::getInstance();
+#endif
+
+#if ANDROID_MAJOR == 4 && ANDROID_MINOR < 2
+    android::Vector<android::String8> matchingCodecs;
+#elif ANDROID_MAJOR < 7
+    android::Vector<android::OMXCodec::CodecNameAndQuirks> matchingCodecs;
+#else
+    android::Vector<android::AString> matchingCodecs;
+#endif
 
-    if (matchIndex >= 0) {
+#if ANDROID_MAJOR < 7
+    android::OMXCodec::findMatchingCodecs(
+            meta->type, encoder, NULL,
+#else
+    android::MediaCodecList::findMatchingCodecs(
+            meta->type, encoder,
+#endif
+            DroidMediaCodecBuilder::flags(meta, 0),
+            &matchingCodecs);
+
+    if (matchingCodecs.size() > 0) {
+        ssize_t matchIndex = list->findCodecByName(matchingCodec.itemAt(0));
```

Comment:
I suggest it should be `matchingCodec.itemAt(0).c_str()`.
For android < 7 this will probably fail: the type of an element is `android::OMXCodec::CodecNameAndQuirks`.

Reply:
It seems that probably is not enough because CodecNameAndQuirks is a struct, I will fix it with some extra ifdefs